src/base/static/components/story-sidebar/index.js

```diff
@@ -0,0 +1,190 @@
+    if (splitUrl.length === 2) {
+      return (
+        this.props.places[
+          this.props.layers.find(layer => layer.slug === splitUrl[0]).id
+        ].get(splitUrl[1]) || {}
+      );
+    } else {
+      // Otherwise, we have a "landmark-style" url, and have to search all place
+      // collections.
+      let foundModel;
+      Object.keys(this.props.places).forEach(collectionName => {
+        let model = this.props.places[collectionName].findWhere({
+          [constants.CUSTOM_URL_PROPERTY_NAME]: splitUrl[0],
+        });
+        if (model) {
+          foundModel = model;
+        }
+      });
+
+      return foundModel || {};
+    }
+  }
+
+  getIconUrl(chapter, storyUrl) {
+    // If the story config for this place declares an explicit sidebar icon url,
+    // use that icon.
+    if (
+      storyConfig[this.state.currentStoryName].order[storyUrl].sidebarIconUrl
+    ) {
+      return storyConfig[this.state.currentStoryName].order[storyUrl]
+        .sidebarIconUrl;
+    }
+
+    // If the story chapter has an icon url defined, use that.
+    if (chapter.get("style") && chapter.get("style").get("iconUrl")) {
+      return chapter.get("style").get("iconUrl");
+    }
+
+    // Otherwise, use the icon url defined in the category config.
+    return placeConfig.place_detail.find(
+      config => config.category === chapter.get("location_type"),
+    ).icon_url;
+  }
+
+  getTitle(chapter) {
+    // This is an unfortunate series of checks, but needed at the moment.
+    // TODO: We should revisit why this is necessary in the first place and see
+    // if we can refactor.
+    return (
+      chapter.get(constants.FULL_TITLE_PROPERTY_NAME) ||
+      chapter.get(constants.TITLE_PROPERTY_NAME) ||
+      chapter.get(constants.NAME_PROPERTY_NAME) ||
+      ""
+    );
+  }
+
+  render() {
+    return (
+      <div className="story-sidebar">
+        <a href="#" className="story-sidebar__collapse-btn" />
+        {this.state.currentStoryHeader && (
+          <h5 className="story-sidebar__header">
+            {this.state.currentStoryHeader}
+          </h5>
+        )}
+        {this.state.currentStoryDescription && (
+          <p className="story-sidebar__description">
+            {this.state.currentStoryDescription}
+          </p>
+        )}
+        <hr />
+        {this.state.currentStoryChapters
+          .map((chapter, placeUrl) => {
+            return (
+              <StoryChapter
+                key={placeUrl}
+                title={this.getTitle(chapter)}
+                iconUrl={this.getIconUrl(chapter, placeUrl)}
+                isSelected={this.state.currentStoryRoute === placeUrl}
+                placeUrl={placeUrl}
+              />
+            );
+          })
+          .toArray()}
+      </div>
+    );
+  }
+}
+
+StorySidebar.propTypes = {
+  layers: PropTypes.array.isRequired,
+  places: PropTypes.object.isRequired,
+  router: PropTypes.instanceOf(Backbone.Router),
```

Comment:
The `Backbone` global is causing an ESLint error. You should add `"Backbone": true,` to our `.eslintrc` to prevent this error.

Keep in mind that we are planning to add an eslint check to our unit tests, so the tests will fail if we get any eslint errors in the code.